src/test/java/ro/msg/learning/shop/service/OrderServiceIntegrationTest.java

```diff
@@ -0,0 +1,97 @@
+package ro.msg.learning.shop.service;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import ro.msg.learning.shop.DummyData;
+import ro.msg.learning.shop.IntegrationTest;
+import ro.msg.learning.shop.exception.LocationSelectionException;
+import ro.msg.learning.shop.model.Address;
+import ro.msg.learning.shop.model.Order;
+import ro.msg.learning.shop.repository.CustomerRepository;
+import ro.msg.learning.shop.repository.LocationRepository;
+import ro.msg.learning.shop.repository.ProductCategoryRepository;
+import ro.msg.learning.shop.repository.ProductRepository;
+import ro.msg.learning.shop.repository.SupplierRepository;
+
+import java.time.LocalDateTime;
+import java.util.Map;
+import java.util.UUID;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+public class OrderServiceIntegrationTest extends IntegrationTest {
+
+    @Autowired
+    private OrderService orderService;
```

Comment:
You already have the @Before method in the IntegrationTest which populate the DB, why do you populate again in the tests?

Reply:
In the IntegrationTest superclass I create only the schema. The tables are not populated. So each test in the @before method can choose which data to save in the database before executing the tests. This way the tests are better isolated from external changes such as the flyway migration scripts and the content of the database is easier to understand while reading the code. Moreover, the integration tests may prepare the database in different ways, therefore this population is not done in the superclass.

Reply:
removed it